src/utils/before-capture.utils.ts

```diff
@@ -0,0 +1,69 @@
+import { clampToRange } from './number-utils'
+import { CaptureResult, KnownEventName } from '../types'
+import { includes } from './index'
+
+function simpleHash(str: string) {
+    let hash = 0
+    for (let i = 0; i < str.length; i++) {
+        hash = (hash << 5) - hash + str.charCodeAt(i) // (hash * 31) + char code
+        hash |= 0 // Convert to 32bit integer
+    }
+    return Math.abs(hash)
+}
+
+function sampleOnProperty(prop: string, percent: number): boolean {
+    return simpleHash(prop) % 100 < clampToRange(percent, 0, 100)
+}
+
+/**
+ * Provides an implementation of sampling that samples based on the distinct ID.
+ * Using the provided percentage.
+ * Can be used to create a beforeCapture fn for a PostHog instance.
+ *
+ * Causes roughly 50% of distinct ids to have events sent.
+ * Not 50% of events for each distinct id.
+ *
+ * @param percent a number from 0 to 100, 100 means never sample, 0 means never send the event
+ */
+export function sampleByDistinctId(percent: number): (c: CaptureResult) => CaptureResult | null {
```

Comment:
I'm wondering if we should have a different folder for these utils (ones we expect customers to use).

I need something similar for a customer, so I'm also wondering where the best place for that would be.

Reply:
Here's my PR https://github.com/PostHog/posthog-js/pull/1517

I put them in a folder called `customizations` which people can import from